src/hotspot/share/ci/ciKlass.cpp

```diff
@@ -226,6 +226,15 @@ jint ciKlass::access_flags() {
   )
 }
 
+// ------------------------------------------------------------------
+// ciKlass::misc_flags
+jint ciKlass::misc_flags() {
+  assert(is_loaded(), "not loaded");
+  GUARDED_VM_ENTRY(
```

Comment:
To Compiler folks: I don't think the VM_ENTRY is necessary, but if it is, then we should consider entering VM mode once and caching/memoizing these immutable flag values in the ciKlass.

Reply:
I agree with Dean. I don't see why the value can't be eagerly captured as part of `ciKlass` initialization.

I'm fine with leaving it as is since it follows the existing pattern in `ciKlass::access_flags()`, so something for a future cleanup.

Reply:
Yes, I agree, this should be saved in ciKlass somewhere, along with access_flags since they are const as well, but I think someone should do this as a follow on.